### PR TITLE
Use certain whiteboard values as labels for the bugtype model

### PR DESCRIPTION
**Pull request by PromiseFru**
The whiteboard tags and their corresponding bug types can be found [here](https://docs.google.com/spreadsheets/d/1WTKFOk0i-gTFV5cYAUJoxh3Dio0KwMC_p0Fpk7E2Oh4/edit?usp=sharing).

Resolves: #720

### CODE REVIEW

File: bugbug/models/bugtype.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def bug_to_types(
 
     if any(
         f"{whiteboard_text}" in bug["whiteboard"].lower()
-        for whiteboard_text in ("overhead", "memshrink")
+        for whiteboard_text in ("overhead", "memshrink", "memshrink:p2")
```
Comment (marco-c) [inline bugbug/models/bugtype.py:42]: `memshrink:p2` is redundant with `memshrink`

File: bugbug/models/bugtype.py [hunk 2]
```diff
@@ -54,6 +54,10 @@ def bug_to_types(
             "snappy",
             "pdfjs-c-performance",
             "pdfjs-performance",
+            "sp3",
+            "fxperf:p3",
+            "fxperf:p1",
+            "fxperf:p2",
```
Comment (marco-c) [inline bugbug/models/bugtype.py:60]: These are reduntant with `fxperf`

**Review by suhaibmujahid — APPROVED** [review]

**Comment by marco-c** [comment]
Could you share the training logs before/after?

**Comment by PromiseFru** [comment]
> Could you share the training logs before/after?

<details>
<summary>Training logs before PR</summary>

```
2023-11-23 19:44:31,021:INFO:__main__:Training *bugtype* model
2023-11-23 19:45:18,543:INFO:bugbug.models.bugtype:7526 crash bugs
2023-11-23 19:45:18,672:INFO:bugbug.models.bugtype:564 memory bugs
2023-11-23 19:45:18,820:INFO:bugbug.models.bugtype:3861 performance bugs
2023-11-23 19:45:18,966:INFO:bugbug.models.bugtype:125 power bugs
2023-11-23 19:45:19,122:INFO:bugbug.models.bugtype:2368 security bugs
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/content/bugbug/scripts/trainer.py", line 161, in <module>
    main()
  File "/content/bugbug/scripts/trainer.py", line 157, in main
    retriever.go(args)
  File "/content/bugbug/scripts/trainer.py", line 51, in go
    metrics = model_obj.train(limit=args.limit)
  File "/content/bugbug/bugbug/model.py", line 356, in train
    self.le.fit(y)
  File "/usr/local/lib/python3.10/dist-packages/sklearn/preprocessing/_label.py", line 98, in fit
    y = column_or_1d(y, warn=True)
  File "/usr/local/lib/python3.10/dist-packages/sklearn/utils/validation.py", line 1156, in column_or_1d
    raise ValueError(
ValueError: y should be a 1d array, got an array of shape (197908, 5) instead.
```
</details>

<details>
<summary>Training logs after PR</summary>

```
2023-11-23 20:17:24,026:INFO:__main__:Training *bugtype* model
2023-11-23 20:18:21,474:INFO:bugbug.models.bugtype:7526 crash bugs
2023-11-23 20:18:21,609:INFO:bugbug.models.bugtype:564 memory bugs
2023-11-23 20:18:21,759:INFO:bugbug.models.bugtype:4388 performance bugs
2023-11-23 20:18:21,917:INFO:bugbug.models.bugtype:125 power bugs
2023-11-23 20:18:22,074:INFO:bugbug.models.bugtype:2598 security bugs
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/content/bugbug/scripts/trainer.py", line 161, in <module>
    main()
  File "/content/bugbug/scripts/trainer.py", line 157, in main
    retriever.go(args)
  File "/content/bugbug/scripts/trainer.py", line 51, in go
    metrics = model_obj.train(limit=args.limit)
  File "/content/bugbug/bugbug/model.py", line 356, in train
    self.le.fit(y)
  File "/usr/local/lib/python3.10/dist-packages/sklearn/preprocessing/_label.py", line 98, in fit
    y = column_or_1d(y, warn=True)
  File "/usr/local/lib/python3.10/dist-packages/sklearn/utils/validation.py", line 1156, in column_or_1d
    raise ValueError(
ValueError: y should be a 1d array, got an array of shape (197908, 5) instead.
```
</details>

😅 I'm not exactly sure why the training isn't completing. I've included the traceback as well. I would appreciate any pointers as to why this might be happening. 🙏

**Review by marco-c — APPROVED** [review]
As we are not running this model in our pipeline, that's probably a bug we have introduced without noticing (so unrelated to your current patch).

The log shows that there are more performance and more security bugs found though, so that shows your changes are correct (and will be useful).

**Pull request merged**